packages/ember-runtime/lib/system/core_object.js

```diff
@@ -125,23 +126,27 @@ function makeCtor() {
               !((keyName === 'actions') && ActionHandler.detect(this))
             );
 
-            let baseValue = this[keyName];
-            let isDescriptor = baseValue !== null && typeof baseValue === 'object' && baseValue.isDescriptor;
+            let possibleDesc = descriptorFor(this, keyName, m);
+            let isDescriptor = possibleDesc !== undefined;
 
-            if (hasConcatenatedProps && concatenatedProperties.indexOf(keyName) > -1) {
-              if (baseValue) {
-                value = makeArray(baseValue).concat(value);
-              } else {
-                value = makeArray(value);
+            if (!isDescriptor) {
```

Comment:
The previous code doesn't do this guard, but it seems strange to put a CP descriptor into a concatenated/merged property. How do we expect these features to interact? Should we assert here?